### PR TITLE
fix #941: implement remove button

### LINKED ISSUE #941: Clicking "Remove Firefox Monitor" doesn't do anything
In the console I see

```
NotFoundError: Node was not found context-buttons.js:60
```

### PR DESCRIPTION
**Pull request by groovecoder**
Okay, this is quite a bit of change and I'd like to get some more eyes on it.

@lesleyjanenorton - can you spot check on this branch a bunch? Sign up/in, add more email addresses, click their unsubscribe links, remove them from preferences, do the full "Remove Firefox Monitor" flow? I tried to do it a bunch but need some more eyes on it.

@pdehaan - also, can you give me a code sanity check here? I hate adding DB-touching code and would like some extra eyes if I'm making mistakes.

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -226,45 +225,73 @@ async function verify(req, res) {
...
+    title: req.fluentFormat("remove-fxm"),
+    subscriber: sessionUser,
+    whichPartial: "subpages/remove_fxm",
+  });
+}
+
+
+async function postRemoveFxm(req, res) {
+  const sessionUser = _requireSessionUser(req);
+  await DB.removeSubscriber(sessionUser);
```
Comment (groovecoder) [inline controllers/user.js]: This definitely needs #961 before launch.

File: controllers/user.js [hunk 2]
```diff
@@ -226,45 +225,73 @@ async function verify(req, res) {
...
+  // Token is for a primary email address,
+  // redirect to preferences to remove Firefox Monitor
+  if (subscriber) {
+    return res.redirect("/user/preferences");
+  }
 
-  //throws error if user backs into and refreshes unsubscribe page
-  if (!subscriber) {
+  const emailAddress = await DB.getEmailByToken(req.query.token);
+  if (!subscriber && !emailAddress) {
```
Comment (pdehaan) [inline controllers/user.js]: Not sure about this logic.
If `subscriber` is truthy on L237-239 above, we `return` and exit this function, which means that the `!subscriber` check is probably always true here. Maybe kinda?

File: public/js/dashboard.js [hunk 3]
```diff
@@ -51,3 +52,11 @@ if (document.querySelector(".email-card")) {
     option.addEventListener("click", sendCommunicationOption);
   });
 }
+
+if (document.querySelector(".remove-fxm")) {
+  const removeMonitorButton = document.querySelector(".remove-fxm");
```
Comment (pdehaan) [inline public/js/dashboard.js]: minor nit, feel free to ignore... but if we move the `const` above the `if`, we can just do:

```js
if (removeMonitorButton) {...}
```

and avoid a second DOM traversal.

File: tests/controllers/user.test.js [hunk 4]
```diff
@@ -258,10 +258,16 @@ test("user unsubscribe GET request with valid token and hash returns 200 without
...
 
-  expect(resp.statusCode).toEqual(200);
+  expect(resp.statusCode).toEqual(302);
+  expect(resp._getRedirectUrl()).toEqual("/user/preferences");
 });
 
 
+// TODO: test("remove-email POST");
+// TODO: test("remove-fxm get");
+// TODO: test("remove-fxm POST");
```
Comment (pdehaan) [inline tests/controllers/user.test.js]: Are you doing these tests in _this_ PR, or is this work for Future Luke(tm)?

Comment (groovecoder) [reply tests/controllers/user.test.js]: Planning to add in _this_ PR.

Comment (groovecoder) [reply tests/controllers/user.test.js]: Now with tests.

File: controllers/user.js (hunk 2 above)
Comment (groovecoder) [reply controllers/user.js]: Good catch.

File: public/js/dashboard.js (hunk 3 above)
Comment (groovecoder) [reply public/js/dashboard.js]: also nice.

**Comment by lesleyjanenorton** [comment]
@groovecoder this looks great! Just went through the various sign up/sign in/delete-this/add-that/unsubscribe-this-secondary-email-from-an-email/etc flows and they all appear to be flowing as expected!

**Comment by groovecoder** [comment]
@pdehaan and/or @lesleyjanenorton - want to double-check my test logic? Make sure I'm testing the right actions with the right DB/fixture users and the right expectations.

**Pull request merged**